CMake fails: Could not find a package configuration file provided by "boost_signals"

@bangerth 
On my Manjaro Linux box, I am trying to install SampleFlow for development. Following the instructions in the README, I executed `cmake .` and got this output: 
```
CMake Error at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR signals)
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.17/Modules/FindBoost.cmake:2145 (find_package_handle_standard_args)
  CMakeLists.txt:34 (FIND_PACKAGE)


-- Configuring incomplete, errors occurred!
See also "/home/dawson/projects/SampleFlow/CMakeFiles/CMakeOutput.log".
```
I figured I just needed to install boost, so I did that and then executed `cmake .` again. Then I got this:
```
CMake Error at /usr/lib64/cmake/Boost-1.72.0/BoostConfig.cmake:120 (find_package):
  Could not find a package configuration file provided by "boost_signals"
  (requested version 1.72.0) with any of the following names:

    boost_signalsConfig.cmake
    boost_signals-config.cmake

  Add the installation prefix of "boost_signals" to CMAKE_PREFIX_PATH or set
  "boost_signals_DIR" to a directory containing one of the above files.  If
  "boost_signals" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /usr/lib64/cmake/Boost-1.72.0/BoostConfig.cmake:185 (boost_find_component)
  /usr/share/cmake-3.17/Modules/FindBoost.cmake:444 (find_package)
  CMakeLists.txt:34 (FIND_PACKAGE)


-- Configuring incomplete, errors occurred!
See also "/home/dawson/projects/SampleFlow/CMakeFiles/CMakeOutput.log".
```
Am I doing something wrong with the installation? Are there other dependencies that need to be installed?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,9 +31,7 @@ INCLUDE_DIRECTORIES(include ${CMAKE_BINARY_DIR}/include)
 
 # Set paths correctly to take BOOST from wherever it is
 # installed on the current system
-FIND_PACKAGE(Boost
-             REQUIRED
-             COMPONENTS signals)
+FIND_PACKAGE(Boost)
 INCLUDE(CheckIncludeFileCXX)
 CHECK_INCLUDE_FILE_CXX("boost/signals2.hpp" SF_HAVE_SIGNALS2)
 IF (NOT SF_HAVE_SIGNALS2)

